runhouse/servers/http/http_server.py

```diff
@@ -699,6 +712,49 @@ def _collect_cluster_stats():
             labels={"username": configs.get("username"), "environment": "prod"},
         )
 
+    @staticmethod
+    def _collect_telemetry_stats():
+        """Collect telemetry stats and send them to the Runhouse hosted OpenTelemetry collector"""
+
+        from opentelemetry import trace
+        from opentelemetry.exporter.otlp.proto.http.trace_exporter import (
+            OTLPSpanExporter,
+        )
+        from opentelemetry.instrumentation.fastapi import FastAPIInstrumentor
+        from opentelemetry.instrumentation.requests import RequestsInstrumentor
+        from opentelemetry.sdk.resources import Resource
+        from opentelemetry.sdk.trace import TracerProvider
+        from opentelemetry.sdk.trace.export import BatchSpanProcessor
+
+        telemetry_collector_address = configs.get("telemetry_collector_address")
+
+        logger.info(f"Preparing to send telemetry to {telemetry_collector_address}")
+
+        # Set the tracer provider and the exporter
+        trace.set_tracer_provider(
+            TracerProvider(
+                resource=Resource.create({"service.name": "runhouse-service"})
```

Comment:
not sure if it's an issue with this tracer or some other setting but the service name does not seem to be propagating into grafana (showing up as "unknown_service")

Reply:
Yes! I've updated the code to have the service and haven't rerun it yet.. The traces that I've posted on Slack were prior to setting the service name.
I'll rerun it again and confirm that it propagates now.

Reply:
Confirmed that service name now shows up:
<img width="1481" alt="Screenshot 2023-12-28 at 11 41 46 AM" src="https://github.com/run-house/runhouse/assets/144945856/b435cb20-4847-431b-b64f-6f40dbee73d4">